packages/@eventual/core/src/workflow.ts

```diff
@@ -136,30 +141,42 @@ export interface Workflow<in Input = any, Output = any> {
  * @param name a globally unique ID for this workflow.
  * @param definition the workflow definition.
  */
-export function workflow<Input = any, Output = any>(
-  name: string,
+export function workflow<
+  Name extends string = string,
+  Input = any,
+  Output = any
+>(
+  name: Name,
   definition: WorkflowHandler<Input, Output>
-): Workflow<Input, Output>;
-export function workflow<Input = any, Output = any>(
-  name: string,
+): Workflow<Name, Input, Output>;
+export function workflow<
+  Name extends string = string,
+  Input = any,
+  Output = any
+>(
+  name: Name,
   opts: WorkflowDefinitionOptions,
   definition: WorkflowHandler<Input, Output>
-): Workflow<Input, Output>;
-export function workflow<Input = any, Output = any>(
+): Workflow<Name, Input, Output>;
+export function workflow<
+  Name extends string = string,
```

Comment:
Do we need it on workflow? Does it have a representation on the cdk side? Maybe monitoring?

Reply:
I think we should be consistent. A workflow may be addressable if it supported overrides to the destination of task, injection of env variables, permission to invoke it, or analytics.

Reply:
Same argument for transactions